h/services/annotation_write.py

```diff
@@ -176,6 +223,54 @@ def _validate_group(self, annotation: Annotation, enforce_write_permission=True)
                 + _("Annotations for this target URI are not allowed in this group")
             )
 
+    def upsert_annotation_slim(self, annotation):
+        self._db.flush()  # See the last model changes in the transaction
+        moderated = self._db.scalar(
```

Comment:
I made this super explicit.

Query the values we need, use the rest from the annotation object.